### PR TITLE
Add DQMGlobalRunSummaryEDAnalyzer to support use of edm::RunSummaryCache

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

Following https://github.com/cms-sw/cmssw/pull/32004#issuecomment-724014288, this PR suggests to add `DQMGlobalRunSummaryEDAnalyzer` to cover the case where `edm::RunSummaryCache` is used with `DQMGlobalEDAnalyzer`. Special treatment is needed because `RunSummaryCache` changes the signature of `globalEndRunProduce()` function.

#### PR validation:

Unit tests in `DQMServices/Demo` pass.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: DQMServices/Demo/test/TestDQMEDAnalyzer.cc [hunk 1]
```diff
@@ -276,6 +276,58 @@ class TestDQMGlobalEDAnalyzer : public DQMGlobalEDAnalyzer<TestHistograms> {
...
+  void streamEndRunSummary(edm::StreamID, edm::Run const&, edm::EventSetup const&, int* runSummaryCache) const {
+    (*runSummaryCache) += 1;
+  }
+
+  void globalEndRunSummary(edm::Run const&, edm::EventSetup const&, int* runSummaryCache) const {}
+
+  void dqmEndRun(edm::Run& run,
+                 edm::EventSetup const& setup,
+                 TestHistograms const& h,
+                 int const& runSummaryCache) const {}
```
Comment (makortel) [inline DQMServices/Demo/test/TestDQMEDAnalyzer.cc:322]: Note that there is some inevitable interface inconsistency: I chose to pass the `RunSummaryCache` object here by reference, while the `streamEndRunSummary` and `globalEndRunSummary` take it by pointer (directly from the framework).

Another question is if users should be required to override `globalEndRunSummary()` (as in here), or should the `DQMGlobalRunSummaryEDAnalyzer` base class provide an empty default implementation (as it does for `dqmEndRun()`.

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by jfernan2** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**